src/kanata/linux.rs

```diff
@@ -202,3 +222,67 @@ fn handle_scroll(
         _ => unreachable!("expect to be handling a wheel event"),
     }
 }
+
+fn start_event_preprocessor(
+    preprocess_rx: Receiver<KeyEvent>,
+    process_tx: Sender<KeyEvent>,
+    debounce_algorithm: Arc<Mutex<Box<dyn Debounce>>>,
+) {
+    std::thread::spawn(move || {
+        {
+            let algorithm = debounce_algorithm.lock();
+            log::info!(
+                "Starting event preprocessor with debounce algorithm: {}, duration: {} ms",
+                algorithm.name(),
+                algorithm.debounce_time()
+            );
+        }
+        let mut last_tick = Instant::now();
+        let mut ms_remainder_in_ns = 0;
+        let mut has_pending_deadlines = false;
+
+        loop {
+            let mut algorithm = debounce_algorithm.lock();
+
+            if has_pending_deadlines {
+                // Tick every 1ms until no pending deadlines
+                let now = Instant::now();
+                let tick_result = count_ms_elapsed(last_tick, now, ms_remainder_in_ns);
+
+                if tick_result.ms_elapsed >= 1 {
+                    // Call tick if at least 1ms has passed
+                    has_pending_deadlines = algorithm.tick(&process_tx, now);
+                    // Update last_tick and remainder based on the helper function's result
+                    last_tick = Instant::now();
+                    ms_remainder_in_ns = 0;
+                }
+
+                // Non-blocking check for new events
+                match preprocess_rx.try_recv() {
+                    Ok(kev) => {
+                        log::debug!("Received event: {:?}", kev);
+                        has_pending_deadlines = algorithm.process_event(kev, &process_tx);
+                    }
+                    Err(TryRecvError::Empty) => {
+                        // No events available, wait a small duration to prevent busy looping
+                        std::thread::sleep(Duration::from_nanos(ms_remainder_in_ns.try_into().unwrap()));
```

Comment:
We could just do a fixed sleep of 1ms here.

Let me know what you think @jtroo

Reply:
Yea always sleeping 1ms is fine here; this is what the processing loop does.